sdgx/models/components/optimize/ndarray_loader.py

```diff
@@ -71,8 +98,12 @@ def get_all(self) -> ndarray:
         return np.concatenate([array for array in self.iter()], axis=1)
 
     @cached_property
+    def __shape_0(self):
+        return self.load(0).shape[0]
+
+    @property
     def shape(self) -> tuple[int, int]:
-        return (self.load(0).shape[0], self.store_index)
+        return self.__shape_0, self.store_index
```

Comment:
Seems like there's no need to do that? `__shape_0` is a little bit confusing as a property

Reply:
If so, `__shape_0` probably shouldn't be cached either?

Reply:
Not this, it's the 'store_index' can't be cached. The shape0 can be cached safety because it's can't be modified after initial.